src/status_im/bottom_sheet/core.cljs

```diff
@@ -17,10 +18,32 @@
    {:view    view
     :options options}))
 
+(rf/defn reset-bottom-sheet
+  {:events [:bottom-sheet/reset]}
+  [{:keys [db]} on-cancel]
+  (when (fn? on-cancel) (on-cancel))
+  {:db (assoc db
+              :bottom-sheet/config
+              {:content-height      nil
+               :show-bottom-sheet?  nil
+               :keyboard-was-shown? false
+               :expanded?           false
+               :gesture-running?    false
+               :animation-delay     constants/bottom-sheet-animation-delay})})
+
 (rf/defn hide-bottom-sheet
   {:events [:bottom-sheet/hide]}
-  [{:keys [db]}]
-  {:db (assoc db :bottom-sheet/show? false)})
+  [{:keys [db]} on-cancel]
```

Comment:
you can do it like this [{:keys [db]} & [{:keys [on-cancel]]]

better to have a map with options instead of just one callback

Reply:
mhm so it won't work with defn macro?

Reply:
btw warning is fine when we use it inside fx/merge

Reply:
yep, sounds good

Reply:
nice thanks

Reply:
the only thing im not sure why `on-cancel` , it  doesn't look like `on-cancel` from code

Reply:
it looks more like `on-did-hide` or `on-hide-finished`